backend/api/api_deployment_views.py

```diff
@@ -74,28 +74,29 @@ def get(
         self, request: Request, org_name: str, api_name: str, api: APIDeployment
     ) -> Response:
         execution_id = request.query_params.get("execution_id")
-        include_metadata = request.query_params.get("include_metadata", False)
+        include_metadata = (
+            request.query_params.get(ApiExecution.INCLUDE_METADATA, "false").lower()
+            == "true"
+        )
         if not execution_id:
             raise InvalidAPIRequest("execution_id shouldn't be empty")
         response: ExecutionResponse = DeploymentHelper.get_execution_status(
             execution_id=execution_id
         )
-        if response.execution_status != CeleryTaskState.SUCCESS.value:
-            return Response(
-                {
-                    "status": response.execution_status,
-                    "message": response.result,
-                },
-                status=status.HTTP_422_UNPROCESSABLE_ENTITY,
-            )
-        if (
-            response.execution_status == CeleryTaskState.SUCCESS.value
-            and not include_metadata
-        ):
-            response.remove_result_metadata_keys()
+        response_status = status.HTTP_200_OK
+        if response.execution_status == CeleryTaskState.COMPLETED.value:
+            if include_metadata:
+                response.remove_result_metadata_keys(keys_to_remove=["highlight_data"])
+            else:
+                response.remove_result_metadata_keys()
```

Comment:
NIT: I could step into the function to understand better - but this gives the impression that in spite of `include_metadata` - we always remove metadata key. See if anything can be renamed / comments added to better improve the understanding

Reply:
@chandrasekharan-zipstack It is clearly mentioned in the doc string of this function. So, I think that it will clear up any confusion an I couldn't come up with any better name 😅.

https://github.com/Zipstack/unstract/blob/830e56382b5725590ccc1ea3f21fa644f62e070e/backend/workflow_manager/workflow/dto.py#L66-L68